plugins/kotlin/core/src/org/jetbrains/kotlin/idea/core/util/inlineAnalysisUtils.kt

```diff
@@ -41,7 +43,10 @@ fun analyzeInlinedFunctions(
         toProcess.add(containingFile)
     }
 
-    return Pair<BindingContext, List<KtFile>>(context, ArrayList(toProcess))
+    // TODO: Can we ensure the analysis results from the underlying project is
+    //       reused/cached to avoid duplicate work?
+    val finalContext = resolutionFacadeForFile.analyzeWithAllCompilerChecks(toProcess)
```

Comment:
Does this correctly reuse and minimize duplicate analysis work?

Reply:
`ResolutionFacade` indeed caches the analysis. So it will not be repeated twice for the same file. However I think it is better to move `resolutionFacadeForFile.analyzeWithAllCompilerChecks(toProcess)` to `org.jetbrains.kotlin.idea.debugger.evaluate.KotlinEvaluator#compileCodeFragment`. The reason is that `org.jetbrains.kotlin.idea.core.util.InlineAnalysisUtilsKt#analyzeInlinedFunctions` is an utility method, it is also used by `org.jetbrains.kotlin.idea.core.KotlinCompilerIde`, and I am a bit worried that changing a binding context and doing some extra analysis may break or slow down something in the future.